code/game/machinery/antiquemattersynth.dm

```diff
@@ -0,0 +1,163 @@
+	density = 1
+	anchored = 0
+	machine_flags = WRENCHMOVE | FIXED2WORK | EMAGGABLE
+	req_access = list(access_engine)
+
+	var/consumption = 0 //How much are we set to draw off the net? Clamped between 0 and 2 GIGAWATT (2,000,000,000 Watts)
+	var/on = 0
+	var/charge = 0 //How much we've stored. Also capped at 2 GIGAWATT.
+	var/category = "resources" //which list to display
+	var/list/categories = list(list("category" = "resources"), list("category" = "tools"), list("category" = "machinery")) //Yes it is necessary to write the list like this
+
+/obj/machinery/power/antiquesynth/proc/toggle_power()
+	on = !on
+	if(!get_powernet())
+		on = FALSE
+		visible_message("<span class='warning'>The [src] buzzes and shuts off.</span>")
+	update_icon()
+
+/obj/machinery/power/antiquesynth/update_icon()
+	return
+	//Maybe I'll add more?
+
+/obj/machinery/power/antiquesynth/process()
+	if(!on)
+		return
+	if(!anchored || !get_powernet())
+		toggle_power()
+		return
+	if(charge >= GIGAWATT)
+		charge = min(charge, GIGAWATT)
+		return //We can't get more charged than this!
+	if(avail()>consumption)
+		charge += consumption
+		add_load(consumption)
+
+/obj/machinery/power/antiquesynth/attack_ai(var/mob/user as mob)
+	to_chat(user, "<span class='warning'>You aren't equipped to interface with technology this old!</span>")
+	return 0
+
+/obj/machinery/power/antiquesynth/attack_paw(var/mob/user as mob)
+	return attack_hand(user)
+
+/obj/machinery/power/antiquesynth/attack_hand(var/mob/user as mob)
+	return ui_interact(user)
+
+/obj/machinery/power/antiquesynth/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open=NANOUI_FOCUS)
+	if (gcDestroyed || !get_turf(src) || !anchored)
+		if(!ui)
+			ui = nanomanager.get_open_ui(user, src, ui_key)
+		if(ui)
+			ui.close()
+		return
+
+	// this is the data which will be sent to the ui
+	var/data[0]
+	data["name"] = name
+	data["powered"] = !(stat & NOPOWER)
+	data["charge"] = charge/MEGAWATT //Charge given in megawatts not watts
+	data["consumption"] = consumption/MEGAWATT //ditto
+	data["active"] = on
+	data["synthList"] = synth_designs
+	data["selectedCategory"] = category
+	data["categories"] = categories
+
+	// update the ui if it exists, returns null if no ui is passed/found
+	ui = nanomanager.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if (!ui)
+		// the ui does not exist, so we'll create a new() one
+        // for a list of parameters and their descriptions see the code docs in \code\\modules\nano\nanoui.dm
+		ui = new(user, src, ui_key, "antiquems.tmpl", "Antique Matter Synthesizer", 520, 460)
+		// when the ui is first opened this is the data it will use
+		ui.set_initial_data(data)
+		// open the new ui window
+		ui.open()
+		// auto update every Master Controller tick
+		//ui.set_auto_update(1)
+
+/obj/machinery/power/antiquesynth/Topic(href, href_list)
+	if(..())
+		return
+	if(!allowed(usr) && !emagged)
+		to_chat(usr,"<span class='warning'>Access denied.</span>")
+		return
+
```

Comment:
This could use sanity checking, maybe check if the user is Adjacent and not incapacitated